### PR TITLE
Update build workflow to include all targets and tests

### PR DESCRIPTION
**Pull request by Nicoretti**
* Make sure all binaries and tests are built
* Fix josh-proxy unittests

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -32,13 +32,15 @@ jobs:
...
       run: cargo fmt -- --check
-    - name: Run Rust tests
-      run: cargo test
+    - name: Run all unit tests
+      run: cargo test --workspace --all
+    - name: Build all targets (libs, binaries, examples, ..)
+      run: cargo build --workspace --all-targets
     - name: Configure environment
       run: |
         export CARGO_TARGET_DIR=`pwd`/target
```
Comment (LMG) [inline .github/workflows/rust.yml:41]: Does this fit with building just before?

Comment (Nicoretti) [reply .github/workflows/rust.yml:41]: I would say it may builds even more e.g. if u have examples etc.

Comment (Nicoretti) [reply .github/workflows/rust.yml:41]: apparently previous build setup did not build josh-proxy test

Comment (LMG) [reply .github/workflows/rust.yml:41]: I mean the CARGO_TARGET_DIR variable.
Looking in detail at the build it looks like it isn't even passed to the script...

File: josh-proxy/Cargo.toml [hunk 2]
```diff
@@ -37,6 +37,6 @@ serde_json= "1.0"
 serde= "*"
 rust-crypto = "0.2"
 unindent = "*"
-juniper = "*"
+juniper = {version = "*", features =["expose-test-schema"] }
```
Comment (christian-schilling) [inline josh-proxy/Cargo.toml:40]: What does that do?

Comment (LMG) [reply josh-proxy/Cargo.toml:40]: exposes the feature which is now hidden behind this. The tests were not running anymore.

**Review by christian-schilling — APPROVED** [review]

**Pull request merged**